app/src/main/java/com/github/palFinderTeam/palfinder/profile/ProfileAdapter.kt

```diff
@@ -76,30 +73,31 @@ class ProfileAdapter(
         val fullName = holder.fullName
         fullName.text = currentDataSet[position].fullName()
         name.text = currentDataSet[position].username
-        CoroutineScope(Dispatchers.IO).launch { currentDataSet[position].pfp.loadImageInto(holder.pic) }
-        if (currentDataSet[position].uuid == loggedUser.uuid) {
-            holder.followButton.visibility = INVISIBLE
-        } else if (canFollow(position)) {
-            holder.followButton.text = holder.view.resources.getString(R.string.follow_button)
-        } else {
-            holder.followButton.text = holder.view.resources.getString(R.string.unfollow_button)
+        when {
+            currentDataSet[position].uuid == loggedUser.uuid -> {
+                holder.followButton.visibility = INVISIBLE
+            }
+            canFollow(position) -> {
+                holder.followButton.text = holder.view.resources.getString(R.string.follow_button)
+            }
+            else -> {
+                holder.followButton.text = holder.view.resources.getString(R.string.unfollow_button)
+            }
         }
         holder.followButton.setOnClickListener {
             if (holder.followButton.text.equals(holder.view.resources.getString(R.string.follow_button))) {
                 follow(position, holder)
             } else {
                 unfollow(position, holder)
             }
-            CoroutineScope(Dispatchers.IO).launch {
-                loggedUser = profileService.fetchUserProfile(loggedUser.uuid)!!
-            }
+        }
 
-            CoroutineScope(Dispatchers.Main).launch {
-                currentDataSet[position].pfp.loadImageInto(
-                    holder.pic,
-                    context
-                )
-            }
+        CoroutineScope(Dispatchers.Main).launch {
+            currentDataSet[position].pfp.loadImageInto(
+                holder.pic,
+                context
+            )
+            holder.pic.visibility = View.VISIBLE
```

Comment:
Setting image visibile become useless with other commit

Reply:
It's because the other branch was merged into it (to simplify conflict solving) but when the other branch will be merged everything should be fine.